packages/std/src/exports.rs

```diff
@@ -44,154 +46,196 @@ extern "C" fn deallocate(pointer: u32) {
     let _ = unsafe { consume_region(pointer as *mut Region) };
 }
 
+// TODO: replace with https://doc.rust-lang.org/std/ops/trait.Try.html once stabilized
+macro_rules! r#try_into_contract_result {
+    ($expr:expr) => {
+        match $expr {
```

Comment:
Can't we just do this with generics rather than macros?

I would generally tend to avoid macros unless they really add value (given this is crate internal, this is better, but still not sure why we use a macro here)

Reply:
We use this e.g. in

```rust
fn _do_init<M, C, E>(
    init_fn: &dyn Fn(
        &mut Extern<ExternalStorage, ExternalApi, ExternalQuerier>,
        Env,
        M,
    ) -> Result<InitResponse<C>, E>,
    env_ptr: *mut Region,
    msg_ptr: *mut Region,
) -> ContractResult<InitResponse<C>>
where
    M: DeserializeOwned + JsonSchema,
    C: Serialize + Clone + fmt::Debug + PartialEq + JsonSchema,
    E: ToString,
{
    let env: Vec<u8> = unsafe { consume_region(env_ptr) };
    let msg: Vec<u8> = unsafe { consume_region(msg_ptr) };
    let env: Env = try_into_contract_result!(from_slice(&env));
    let msg: M = try_into_contract_result!(from_slice(&msg));
    let mut deps = make_dependencies();
    init_fn(&mut deps, env, msg).into()
}
```

because the `?` operator is not available for `ContractResult`